staging/src/k8s.io/kube-gen/cmd/conversion-gen/generators/conversion.go

```diff
@@ -329,6 +329,13 @@ func (e equalMemoryTypes) Skip(a, b *types.Type) {
 }
 
 func (e equalMemoryTypes) Equal(a, b *types.Type) bool {
+	// alreadyVisitedTypes holds all the types that have been converted. This is used to avoid
```

Comment:
here it's not about "been converted", but about comparing that two structs have the same memory structure. If they have, we can use `unsafe.Pointer` to replace conversion by a type cast.

Better: 
```
// alreadyVisitedTypes holds all the types that have already been checked in the structural type recursion.
```.